R/human_infection.R

```diff
@@ -155,21 +108,107 @@ calculate_infections <- function(
   }
 
   prob <- b * (1 - prophylaxis) * (1 - vaccine_efficacy)
-  infected <- bitset_at(source_humans, bernoulli_multi_p(prob))
 
-  incidence_renderer(
+  ## probability of incidence must be rendered at each timestep
+  incidence_probability_renderer(
     variables$birth,
     renderer,
-    infected,
-    source_humans,
+    source_pop,
     prob,
+    "inc_",
+    parameters$incidence_min_ages,
+    parameters$incidence_max_ages,
+    timestep
+  )
+  
+  ## capture infection rates to resolve in competing hazards
+  infection_rates <- numeric(length = parameters$human_population)
+  infection_rates[source_vector] <- prob_to_rate(prob)
+  infection_outcome$set_rates(infection_rates)
+}
+
+#' @title Assigns infections to appropriate human states
+#' @description
+#' Updates human states and variables to represent asymptomatic/clinical/severe
+#' and treated malaria; and resulting boosts in immunity
+#' @param timestep current timestep
+#' @param infected_humans bitset of infected humans
+#' @param source_humans bitset of humans with infection potential
+#' @param variables a list of all of the model variables
+#' @param renderer model render object
+#' @param parameters model parameters
+#' @param prob vector of population probabilities of infection
+#' @noRd
+infection_process_resolved_hazard <- function(
```

Comment:
Just a suggestion, this feels easier to understand

```suggestion
infection_outcome_process <- function(
```